src/Game.js

```diff
@@ -0,0 +1,80 @@
+class Game {
+  constructor(frame = new Frame()) {
+    this.frameNo = 1;
+    this.frames = [];
+    this.currentFrame = frame;
+    this.currentscore = 0;
+    this.gameOver = false;
+  }
+
+  nextFrame(frame = new Frame()) {
+    if (this.frameNo > 10) {
+      return (this.gameOver = true);
+    } else {
+      this.frames.push(this.currentFrame);
+      this.frameNo++;
+      if (this.frameNo < 10) {
+        this.currentFrame = frame;
+      } else if (this.frameNo === 10) {
+        this.finalFrame();
+      }
+    }
+  }
+
+  finalFrame(frame = new FinalFrame()) {
+    this.currentFrame = frame;
+  }
+
+  rollBall(value) {
+    this.currentFrame.roll(value);
+    if (this.currentFrame.endFrame) {
+      this.nextFrame();
+    }
+  }
+
+  score() {
```

Comment:
This score method is quite large - may want to extract some code out to separate methods and just call them in this method instead?